### PR TITLE
clarify BLEPeripheral constructor pin required, but hardcoded/ignored…

### PR DESCRIPTION
**Pull request by jacobrosenthal**
… for some boards

Happy to argue verbiage

### CODE REVIEW

**Comment by sandeepmistry** [comment]
Maybe we should just remove the pin stuff, and add a comment about being able to customize them for nRF8001.

I added a default args in https://github.com/sandeepmistry/arduino-BLEPeripheral/commit/6e758dc7de3e95b62527f37612c8374a9cb4a3b2 to match CurieBLE.

**Comment by jacobrosenthal** [comment]
With those new (old) defaults already in, and nrf8001 style boards going out of favor and nrf51/52 boards the current standard I see just removing the comments from examples entirely and adding a readme footnote?
```
//   Adafruit Bluefruit LE   10, 2, 9
//   Blend                    9, 8, UNUSED
//   Blend Micro              6, 7, 4
//   RBL BLE Shield           9, 8, UNUSED
```

The default constructor is currently filled by these pins, and thus we've lost the  9, 8, UNUSED configuration
```
#if defined(NRF51) || defined(NRF52) || defined(__RFduino__)
  #define BLE_DEFAULT_REQ   -1
  #define BLE_DEFAULT_RDY   -1
  #define BLE_DEFAULT_RST   -1
#elif defined(BLEND_MICRO)
  #define BLE_DEFAULT_REQ   6
  #define BLE_DEFAULT_RDY   7
  #define BLE_DEFAULT_RST   4
#else
  #define BLE_DEFAULT_REQ   10
  #define BLE_DEFAULT_RDY   2
  #define BLE_DEFAULT_RST   9
#endif
```

BLEND_MICRO isnt defined anywhere in arduino-nRF5, Looks like its coming from Redbear board definitions, https://github.com/RedBearLab/Blend/tree/master/Arduino/hardware/blend/avr/variants
```
#define BLEND
```
and
```
#define BLEND_MICRO
#define BLEND_MICRO_16MHZ
```

And implemented in their nrf8001 library as 
https://github.com/RedBearLab/nRF8001/blob/fd3d32212fae78ef717bceea9c5c87a493d9b29d/src/RBL_nRF8001.h#L31-L37
```
#if defined(BLEND_MICRO)
#define DEFAULT_REQN    6
#define DEFAULT_RDYN    7
#else
#define DEFAULT_REQN    9
#define DEFAULT_RDYN    8
#endif
```

NOTE I dont see a #define there for RBL BLE Shield nor can I find it elsewhere, I think were just going to have to lose it.

I also can cant find any information on their reset pin, doesnt seem to be hardcoded, they have an accessor function that takes a reset pin as argument, but none of their examples call it:
https://github.com/RedBearLab/nRF8001/blob/fd3d32212fae78ef717bceea9c5c87a493d9b29d/src/RBL_nRF8001.cpp#L244


Im just going to cargo cult the BLEND_MICRO reset pin in our definitions for BLEND like:
```
#elif defined(BLEND)
  #define BLE_DEFAULT_REQ   9
  #define BLE_DEFAULT_RDY   8
  #define BLE_DEFAULT_RST   4
```

Updated pr soon

**Comment by jacobrosenthal** [comment]
Needs testing and review still

File: API.md [hunk 1]
```diff
@@ -2,8 +2,7 @@
 
 ## Constructor
 ```c
-BLEPeripheral(unsigned char req, unsigned char rdy, unsigned char rst);
-```
+BLEPeripheral(unsigned char req = BLE_DEFAULT_REQ, unsigned char rdy = BLE_DEFAULT_RDY, unsigned char rst = BLE_DEFAULT_RST);
  * req - REQ pin
  * rdy - RDY pin
  * rst - RST pin, can be ```UNUSED```
```
Comment (jacobrosenthal) [inline API.md]: Should we clarify further that these are defaults and can be left out? If so, what verbage?

Comment (sandeepmistry) [inline API.md]: Right, probably need to refer to https://github.com/sandeepmistry/arduino-BLEPeripheral#pinouts

so just a link?

File: src/BLEPeripheral.h [hunk 2]
```diff
@@ -34,12 +34,17 @@
   #define BLE_DEFAULT_REQ   6
   #define BLE_DEFAULT_RDY   7
   #define BLE_DEFAULT_RST   4
+#elif defined(BLEND)
+  #define BLE_DEFAULT_REQ   9
```
Comment (sandeepmistry) [inline src/BLEPeripheral.h]: This looks good, not sure why I left it out :)

**Comment by sandeepmistry** [comment]
I think this is good to merge once we sort out https://github.com/sandeepmistry/arduino-BLEPeripheral/pull/130#discussion_r106556493

File: API.md (hunk 1 above)
Comment (sandeepmistry) [inline API.md]: @jacobrosenthal ping, do you think we should add the link?

After that, let's merge this!

**Pull request merged**